### PR TITLE
Get the cmd line from procfs before calling ps -p

### LINKED ISSUE #75: Don't depend on `ps` supporting `-p`
Hi @Nikratio 

First of all, many thanks for this incredibly useful project! I'm currently doing some experiments running s3ql on Alpine Linux / BusyBox. Basically, this seems to work just fine, but there's one little caveat: `umount.s3ql` is relying on [calling `ps -p`](https://github.com/s3ql/s3ql/blob/master/src/s3ql/umount.py#L94), which is [not supported in the BusyBox `ps` implementation](https://busybox.net/downloads/BusyBox.html#ps). Since this does not work and `umount.s3ql` then [just assumes that the daemon has stopped](https://github.com/s3ql/s3ql/blob/master/src/s3ql/umount.py#L97), I have to make sure myself that the main s3ql daemon has really finished after calling `umount.s3ql` (e.g., by calling `wait <pid>` after calling `umount.s3ql`).

My first thought was that getting the command line could also be accomplished with looking at `/proc/<PID>/cmdline`, but I guess that would not be compatible with OS X anymore? Maybe umount.s3ql could first look for `/proc` and only use `ps -p` as a backup if no procfs is available?

Alternatively, I would request to redirect stderr of the `ps` call to `/dev/null`, so BusyBox does at least not clutter my s3ql logs with log messages saying it does not support the `-p` flag.

### PR DESCRIPTION
**Pull request by stjosh**
Fixes #75

### CODE REVIEW

**Comment by stjosh** [comment]
Closing - this PR does not work well if there is a `/proc` filesystem, but the process has already finished

**Comment by stjosh** [comment]
Reopening with a proper implementation

**Comment by stjosh** [comment]
@Nikratio So here's my PR for handling #75 . Thanks for considering a merge.

File: src/s3ql/umount.py [hunk 1]
```diff
@@ -89,14 +89,27 @@ def get_cmdline(pid):
...
+    
+    output = None
+    if os.path.isdir('/proc'):
+        cmd_path = '/proc/%d/cmdline' % pid
+        if os.path.isfile(cmd_path):
+            try:
+                with open(cmd_path, 'r') as cmd_file:
+                    output = cmd_file.read()
+
+            except Exception:
```
Comment (Nikratio) [inline src/s3ql/umount.py]: No catch-all exception handlers please. I believe what you want to catch is only FileNotFound, which actually makes the `isfile` (or `exists`) check superfluous.

File: src/s3ql/umount.py [hunk 2]
```diff
@@ -89,14 +89,27 @@ def get_cmdline(pid):
...
     and return None.
     '''
+    
+    output = None
+    if os.path.isdir('/proc'):
+        cmd_path = '/proc/%d/cmdline' % pid
+        if os.path.isfile(cmd_path):
+            try:
+                with open(cmd_path, 'r') as cmd_file:
+                    output = cmd_file.read()
```
Comment (Nikratio) [inline src/s3ql/umount.py]: Why not simply `return cmd_file.read()`?

File: src/s3ql/umount.py [hunk 3]
```diff
@@ -89,14 +89,27 @@ def get_cmdline(pid):
...
+
+            except Exception:
+                log.warning('Error when reading from procfs, assuming process %d has terminated.'
```
Comment (Nikratio) [inline src/s3ql/umount.py]: This is racy. The process may terminate between the `isfile` check and the opening, then you'd still get a spurious error message. So I would drop the `isfile` check completely, catch only the `FileNotFound` exception and (when caught) return `None` without logging an error.

File: src/s3ql/umount.py (hunk 2 above)
Comment (stjosh) [reply src/s3ql/umount.py]: Well, I saw cases on my own system where `/proc/<pid>/cmdline` was empty (I guess that's for early bootstrapping processes?). I thought it would make sense to handle this case and return `None` instead of an empty string. 

But now, when looking at the code again how `get_cmdline` is being used, I guess it actually makes more sense to return the empty string in that case and only return `None` in case the cmdline could not be determined if either the process does not exist anymore or in case of an error.

TL;DR: Changed accordingly.

File: src/s3ql/umount.py (hunk 1 above)
Comment (stjosh) [reply src/s3ql/umount.py]: I see. Changed.

File: src/s3ql/umount.py (hunk 3 above)
Comment (stjosh) [reply src/s3ql/umount.py]: Completely agree, thanks! Changed.

**Comment by stjosh** [comment]
Many thanks for your comments - I guess you have noticed that python is not my "home"-language. :) So, I learned another bit of good style and habits and hope the latest version is good to merge.
